### PR TITLE
A) Move to Boto3(decrease memory usage), B) Allow Role Authentication

### PR DESCRIPTION
**Pull request by raphael-riel**
## Move to Boto3(decrease memory usage)

With my large vaults (500k+ Archives), I was unable to delete the whole vault without running in a memory depletion. Even on a `r4.2xlarge` EC2 machine from AWS (61GB RAM).

With some research, it was looking like boto was leaking on each request.
With the move toward boto3, the same run takes no more than ~400Mb.

~Caveat: Boto3 needed AWS's account id. It's now required in the run args.~

## Allow role Authentication

I needed the script to run on an EC2 with access to Glacier via a IAM Role.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
     # Project uses reStructuredText, so ensure that the docutils get
     # installed or upgraded on the target machine
-    install_requires = ['boto>=2.9.9'],
+    install_requires = ['3>=1.4.4'],
```
Comment (spg) [inline setup.py]: shouldn't it be `boto3>=1.4.4`?

Comment (raphael-riel) [inline setup.py]: You're right.

**Comment by spg** [comment]
LGTM

File: removeVault.py [hunk 2]
```diff
@@ -31,7 +34,10 @@ def process_archive(archive_list):
 
 				logging.info('Retry to remove archive ID : %s', archive['ArchiveId'])
 				try:
-					vault.delete_archive(archive['ArchiveId'])
+					glacier.delete_archive(
+					    vaultName='vaultName',
```
Comment (leeroybrun) [inline removeVault.py]: Shouldn't it be `vaultName=vaultName`, without quotes?

Comment (raphael-riel) [inline removeVault.py]: Ahah! Yopu're right! I had to fix this while testing on my EC2 and fogot to apply the fix in the git repo. Thanks!

Comment (leeroybrun) [inline removeVault.py]: Great, thanks! :)

**Comment by raphael-riel** [comment]
I just found out I could dynamically fetch the AccountID using Boto...
http://stackoverflow.com/questions/36709461/get-aws-account-id-from-boto
I'll apply this method in a few moments.

**Comment by leeroybrun** [comment]
Looks good! Thanks a lot for your work! :)

**Comment by raphael-riel** [comment]
Thanks to you too for initialy sharing your work. This saved me a big headache!

**Pull request merged**